src/formatting_stack/reporters/impl.clj

```diff
@@ -1,12 +1,13 @@
 (ns formatting-stack.reporters.impl
   (:require
    [clojure.string :as string]
+   [medley.core :refer [assoc-some]]
    [nedap.speced.def :as speced]))
 
 (speced/defn normalize-filenames
   "Removes the CWD from the filenames, for more concise output, and also for ensuring correct grouping."
-  [{:keys [^string? filename] :as report}]
-  (assoc report :filename (-> filename
-                              (string/replace (re-pattern (str "^" (System/getProperty "user.dir")))
-                                              "")
-                              (string/replace #"^/" ""))))
+  [{:keys [^::speced/nilable ^string? filename] :as report}]
```

Comment:
wdyt of `def-with-doc`ing `^::speced/nilable ^string?` in a centralized spec?

That way we document the nuance.

It's also a good opportunity to review (and possibly upgrade) other specings of `filename`

Reply:
`:formatting-stack.protocols.spec/report` has 2 specs, one where `:filename` is required, and one for exceptions where `:filename` is optional, so the it _is_ documented. But it would be fine in this case to remind the reader.

Reply:
>  it is documented

But ` ^::speced/nilable ^string? filename` does not relate to those 2 specs (b/c it's inline), so one cannot necessarily guess that relationship